File: server/node/subscribe-user-to-list.js
import Courier from '@trycourier/courier';
import dotenv from 'dotenv';

dotenv.config({ path: new URL('../.env', import.meta.url) });

const apiKey = process.env.COURIER_API_KEY;
const listId = process.env.COURIER_SUBSCRIBE_USER_TO_LIST_LIST_ID;
const userId = process.env.COURIER_SUBSCRIBE_USER_TO_LIST_USER_ID;

const client = new Courier({ apiKey });

await client.lists.subscriptions.subscribeUser(userId, {
  list_id: listId,
  preferences: {
    categories: {},
    notifications: {}
  }
});

File: server/python/subscribe-user-to-list.py
import os
from pathlib import Path
from dotenv import load_dotenv
from courier import Courier

load_dotenv(Path(__file__).parent.parent / ".env")

api_key = os.getenv("COURIER_API_KEY")
list_id = os.getenv("COURIER_SUBSCRIBE_USER_TO_LIST_LIST_ID")
user_id = os.getenv("COURIER_SUBSCRIBE_USER_TO_LIST_USER_ID")

client = Courier(api_key=api_key)

client.lists.subscriptions.subscribe_user(
    list_id=list_id,
    user_id=user_id,
)

